experimental/QuadFormAndIsom/src/enumeration.jl

```diff
@@ -1148,8 +1170,21 @@ function splitting_of_hermitian_type(
       end
       isempty(Bs) && continue
       for LA in As, LB in Bs
-        Es = admissible_equivariant_primitive_extensions(LA, LB, Lf, p; check=false)
-        append!(reps, Es)
+        Es = admissible_equivariant_primitive_extensions(LA, LB, Lf, p; check=false, test_type=false)
+        if fix_root == k
+          while !isempty(Es)
+            M = pop!(Es)
+            j = findfirst(l -> is_of_same_type(M^(l*p), Lf), Sk)
+            if isnothing(j)
+              continue
+            end
+            l = Sk[j]
+            push!(reps, M^l)
+          end
+        else
+          filter!(M -> is_of_same_type(M^p, Lf), Es)
```

Comment:
Aside: I find the notation `M^p` to denote $(M,f_M^p)$ slightly disturbing.

Reply:
We have not moved the project out of experimental yet, we can still change the notation/function's name if you have any suggestion.